### DISCUSSION TITLE
On bundling angular modules

### DISCUSSION THREAD

**Discussion by qortex** [ChristianKohler/Homepage#21] [Blog Posts]
Hi,

Thanks for the awesome writeup! https://christiankohler.net/bundling-angular-modules

I've been wondering how it works for ages! Very clear explanation and satisfying to see the process is so nicely automatically optimized.

I'm still wondering how services are bundled though. Let's say I have a service provided in `'root'` in a lazy module: how is it bundled? Actually, asking that, I'm not sure what the behaviour is if a component outside the lazy module tries to use the service. Is the mechanism clever enough to drag the service in the main bundle, or does it fail because the module is not loaded yet (and thus the provider is not yet available)? Or maybe it means all root services are bundled in the main bundle to avoid that?

I guess my question is: components, directive & pipes are pretty direct to pinpoint because they are declared in a specific `NgModule` definition. But since services are not, is there anything specific to know about where they are bundled?

**Comment by qortex** [comment]
This article covers the topic very well! : https://juristr.com/blog/2021/04/angular-di-and-lazy-modules/

Maybe a link to it would be useful in the blog post :)

**Reply by ChristianKohler** [reply]
Thank you for the input. Added the link to Juris blog. Indeed a very good explanation.
